test/hotspot/jtreg/vmTestbase/nsk/jvmti/InterruptThread/intrpthrd003.java

```diff
@@ -51,71 +53,37 @@ public static void main(String[] args) {
         System.exit(run(args, System.out) + 95/*STATUS_TEMP*/);
     }
 
-    public static int run(String argv[], PrintStream ref) {
-        intrpthrd003a runn[] = new intrpthrd003a[THREADS_NUMBER];
-
-        System.out.println("Case 1: JVM/TI InterruptThread()");
-        for (int i = 0; i < THREADS_NUMBER; i++ ) {
-            runn[i] = new intrpthrd003a();
-            int late_count = 1;
-            synchronized (runn[i].syncObject) {
-                runn[i].start();
-                try {
-                    runn[i].syncObject.wait();
-
-                    for (; late_count <= N_LATE_CALLS; late_count++) {
-                        if (check(i, runn[i]) == 2) break;
-
-                        if (isThreadNotAliveError()) {
-                            // Done with InterruptThread() calls since
-                            // thread is not alive.
-                            break;
-                        }
-                    }
-                } catch (InterruptedException e) {
-                    throw new Error("Unexpected: " + e);
-                }
-            }
-
-            System.out.println("INFO: thread #" + i + ": made " + late_count +
-                               " late calls to JVM/TI InterruptThread()");
-            System.out.println("INFO: thread #" + i + ": N_LATE_CALLS==" +
-                               N_LATE_CALLS + " value is " +
-                               ((late_count >= N_LATE_CALLS) ? "NOT " : "") +
-                               "large enough to cause an InterruptThread() " +
-                               "call after thread exit.");
-
+    public static int run(String args[], PrintStream ref) {
+        int timeMax = 0;
+        if (args.length == 0) {
+            timeMax = DEF_TIME_MAX;
+        } else {
             try {
-                runn[i].join();
-            } catch (InterruptedException e) {
-                throw new Error("Unexpected: " + e);
-            }
-            if (check(i, runn[i]) == 2) break;
-            if (!isThreadNotAliveError()) {
-                throw new Error("Expected JVMTI_ERROR_THREAD_NOT_ALIVE " +
-                                "after thread #" + i + " has been join()'ed");
+                timeMax = Integer.parseUnsignedInt(args[0]);
+            } catch (NumberFormatException nfe) {
+                System.err.println("'" + args[0] + "': invalid timeMax value.");
+                    usage();
             }
         }
 
-        int res = getResult();
-        if (res != 0) {
-            return res;
-        }
+        System.out.println("About to execute for " + timeMax + " seconds.");
+
+        long count = 0;
+        long start_time = System.currentTimeMillis();
+        while (System.currentTimeMillis() < start_time + (timeMax * 1000)) {
+            count++;
 
-        System.out.println("Case 2: java.lang.Thread.interrupt()");
```

Comment:
Was Case 2 removed because it is covered by runtime/Thread/InterruptAtExit.java

Reply:
Yes, it seemed like duplicated coverage to me.